services/src/main/java/org/exoplatform/addons/gamification/service/mapper/DomainMapper.java

```diff
@@ -26,86 +25,86 @@
 
-    public DomainDTO domainToDomainDTO(DomainEntity domain) {
-        return new DomainDTO(domain);
-    }
+  public DomainMapper() {
+  }
 
-    public List<DomainDTO> domainssToDomainDTOs(List<DomainEntity> domains) {
-        return domains.stream()
-                .filter(Objects::nonNull)
-                .map((DomainEntity domain) -> domainToDomainDTO(domain))
-                .collect(Collectors.toList());
-    }
+  public static DomainDTO domainToDomainDTO(DomainEntity domain) {
+    return new DomainDTO(domain);
+  }
 
-    public DomainEntity domainDTOToDomain(DomainDTO domainDTO) {
+  public static List<DomainDTO> domainssToDomainDTOs(List<DomainEntity> domains) {
+    return domains.stream()
+                  .filter(Objects::nonNull)
+                  .map((DomainEntity domain) -> domainToDomainDTO(domain))
+                  .collect(Collectors.toList());
+  }
 
-        if (domainDTO == null) {
-            return null;
-        } else {
-            DomainEntity domain = new DomainEntity();
-            domain.setId(domainDTO.getId());
-            domain.setTitle(domainDTO.getTitle());
-            domain.setDescription(domainDTO.getDescription());
-            domain.setCreatedBy(domainDTO.getCreatedBy());
-            domain.setLastModifiedBy(domainDTO.getLastModifiedBy());
-            domain.setDeleted(domainDTO.isDeleted());
-            domain.setEnabled(domainDTO.isEnabled());
-            try {
-                domain.setCreatedDate(formatter.parse(domainDTO.getCreatedDate()));
-                domain.setLastModifiedDate(formatter.parse(domainDTO.getLastModifiedDate()));
-            } catch (Exception exception) {
-                // Date format unknown, or missing properties, we can ignore
-            }
-            domain.setPriority(domainDTO.getPriority());
-            return domain;
-        }
-    }
+  public static DomainEntity domainDTOToDomain(DomainDTO domainDTO) {
 
-    public List<DomainEntity> domainDTOsToDomains(List<DomainDTO> domainDTOs) {
-        return domainDTOs.stream()
-                .filter(Objects::nonNull)
-                .map((DomainDTO domainDTO) -> domainDTOToDomain(domainDTO))
-                .collect(Collectors.toList());
+    if (domainDTO == null) {
+      return null;
+    } else {
+      DomainEntity domain = new DomainEntity();
+      domain.setId(domainDTO.getId());
+      domain.setTitle(domainDTO.getTitle());
+      domain.setDescription(domainDTO.getDescription());
+      domain.setCreatedBy(domainDTO.getCreatedBy());
+      domain.setLastModifiedBy(domainDTO.getLastModifiedBy());
+      domain.setDeleted(domainDTO.isDeleted());
+      domain.setEnabled(domainDTO.isEnabled());
+      try {
+        domain.setCreatedDate(formatter.parse(domainDTO.getCreatedDate()));
+        domain.setLastModifiedDate(formatter.parse(domainDTO.getLastModifiedDate()));
+      } catch (Exception exception) {
+        // Date format unknown, or missing properties, we can ignore
+      }
```

Comment:
Delete try catch && use DateTimeFormatter in Utils and add a check on the date if null or not